### PR TITLE
refactor[docs]: restructure the documentation

### PR DESCRIPTION
**Pull request by sebastianpfischer**

### CODE REVIEW

**Comment by sebastianpfischer** [comment]
I think you should clone the repo to get a feeling of the changes

File: docs/advanced_usage/advanced_tests.rst [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+               else: # Do not forget to send a setup message to the other auxiliaries!
+                   # Compose the normal message
+                   testcase_setup_basic_message = message.Message(msg_type=message.MessageType.COMMAND, sub_type=message.MessageCommandType.TEST_CASE_SETUP,
+                                                           test_section=self.test_section_id, test_suite=self.test_suite_id, test_case=self.test_case_id)
+                   # Send the message
+                   aux.run_command(testcase_setup_basic_message, blocking=True, timeout_in_s=10)
+
+.. _implementation-of-the-tests-advance-1:
+
+Implementation of Advanced Tests - Test Templates
```
Comment (sebclrsn) [inline docs/advanced_usage/advanced_tests.rst:272]: We should also at some point explain the general test flow, e.g.: 
```
SuiteSetup().test_suite_setUp 
TestCase1.setUpClass
TestCase1().setUp
TestCase1().test_run
TestCase1().tearDown
TestCase1().setUp
TestCase1().test_run_2
TestCase1().tearDown
TestCase1.tearDownClass
TestCase2.setUpClass
TestCase2().setUp
TestCase2().test_run
TestCase2().tearDown
TestCase2().setUp
TestCase2().test_run_2
TestCase2().tearDown
TestCase2.tearDownClass
TestSuite().test_suite_tearDown
```

Comment (sebastianpfischer) [reply docs/advanced_usage/advanced_tests.rst:182]: I added it in basic_tests.rst (the end)

**Review by Pog3k — APPROVED** [review]

File: docs/tools/index.rst [hunk 2]
```diff
@@ -1,5 +1,8 @@
-Provided tools
-==============
+Additional Tools
+================
+
+When some tasks are repetitive, the implementation of a tool to do them are needed.
```
[the remark this answers is not in the document]
Comment (sebastianpfischer) [reply docs/tools/index.rst]: It's true

**Pull request merged**